### ISSUE TITLE
Review all test cases

### ISSUE THREAD

**Issue by sudar** [sudar/bulk-move#84]

**Comment by mariadanieldeepak** [comment]
@sudar,

The following tests are unnecessary as the User wouldn't be able to emulate these scenarios.

https://github.com/sudar/bulk-move/blob/dev/2.0.0/tests/wp-tests/BM/Metabox/Posts/CategoryTest.php#L26-L57

https://github.com/sudar/bulk-move/blob/dev/2.0.0/tests/wp-tests/BM/Metabox/Posts/CategoryTest.php#L147-L181

https://github.com/sudar/bulk-move/blob/dev/2.0.0/tests/wp-tests/BM/Metabox/Posts/CategoryTest.php#L271-L305

https://github.com/sudar/bulk-move/blob/dev/2.0.0/tests/wp-tests/BM/Metabox/Posts/CategoryTest.php#L395-L421

https://github.com/sudar/bulk-move/blob/dev/2.0.0/tests/wp-tests/BM/Metabox/Posts/TagTest.php#L21-L58

https://github.com/sudar/bulk-move/blob/dev/2.0.0/tests/wp-tests/BM/Metabox/Posts/TagTest.php#L21-L58

https://github.com/sudar/bulk-move/blob/dev/2.0.0/tests/wp-tests/BM/Metabox/Posts/TagToCategoryTest.php#L21-L57

https://github.com/sudar/bulk-move/blob/dev/2.0.0/tests/wp-tests/BM/Metabox/Posts/CategoryToTagTest.php#L21-L60

The following scenarios are not covered in the tests.

![image](https://user-images.githubusercontent.com/17510033/36387679-9ee191c0-15bf-11e8-82ac-61d8ed23d1c1.png)

**Clarification**
How can the Users select the option to move posts from the default Category? (Does it represent the Uncategorized category?)

https://github.com/sudar/bulk-move/blob/dev/2.0.0/tests/wp-tests/BM/Metabox/Posts/CategoryTest.php#L183-L494

**Comment by sudar** [comment]
@mariadanieldeepak 

> The following tests are unnecessary as the User wouldn't be able to emulate these scenarios.

Can you explain this a bit more?

> How can the Users select the option to move posts from the default Category? (Does it represent the Uncategorized category?)

Posts should belong to atleast one category. If we remove all categories from the post then they will be assigned to the default category.

**Comment by mariadanieldeepak** [comment]
@sudar,

> Can you explain this a bit more?

Consider the following case for example (Move Posts by Category). The User cannot avoid the filter at any cause. One of the filter options is selected by default.

![image](https://user-images.githubusercontent.com/17510033/36406916-5b431554-1620-11e8-9225-b3251e4263ab.png)

So these tests make more sense to me

```
test_move_posts_from_one_cat_to_another_with_overwrite
test_move_posts_from_one_cat_to_another_without_overwrite
```

than this

`test_move_posts_from_one_cat_to_another`

Posts should belong to atleast one category. If we remove all categories from the post then they will be assigned to the default category.

> Posts should belong to atleast one category. If we remove all categories from the post then they will be assigned to the default category.

Alright, got it.

**Comment by sudar** [comment]
@mariadanieldeepak 

If we set `'overwrite' => true`, then it means we are choosing the first option in the filter and if we set to false then we are choosing the second option in the filter.

**Comment by mariadanieldeepak** [comment]
@sudar,

> If we set 'overwrite' => true, then it means we are choosing the first option in the filter and if we set to false then we are choosing the second option in the filter.

Absolutely. By default one of the overwrite filter is always set. That is why the following test case doesn't really add up (at least to me or I could be totally de-railed).

`test_move_posts_from_one_cat_to_another`

**Comment by mariadanieldeepak** [comment]
@sudar,

The missing test cases have been identified and an [issue](https://github.com/sudar/bulk-move/issues/86) has been created for the same.

This issue can be closed now, in case you don't have anything to add. Should we create another issue to remove the repetitive test cases?

**Comment by sudar** [comment]
@mariadanieldeepak 

Let's create a new issue to remove all the duplicate test cases and assign it to @anandB000 

Once it is done, we can close this issue.

**Comment by mariadanieldeepak** [comment]
@sudar,

Issue #89 has been created to address removal of repetitive test cases. Hence I'm closing this issue.
